scripts/variantstore/wdl/GvsQuickstartIntegration.wdl

```diff
@@ -0,0 +1,224 @@
+        # Dataset names must be alphanumeric and underscores only. Convert any dashes to underscores, then delete
+        # any remaining characters that are not alphanumeric or underscores.
+        dataset="$(echo quickit_${branch}_${hash} | tr '-' '_' | tr -c -d '[:alnum:]_')"
+
+        bq mk --project_id="spec-ops-aou" "$dataset"
+
+        echo -n "$dataset" > dataset.txt
+    >>>
+
+    output {
+        File jar = glob("gatk/build/libs/*-SNAPSHOT-local.jar")[0]
+        String dataset_name = read_string("gatk/dataset.txt")
+    }
+
+    runtime {
+        docker: "gcr.io/google.com/cloudsdktool/cloud-sdk:latest"
+        disks: "local-disk 500 HDD"
+    }
+}
+
+task AssertIdenticalOutputs {
+    input {
+        String expected_output_prefix
+        Array[File] actual_vcfs
+    }
+
+    command <<<
+        failures=()
+
+        # Where the current set of expected results lives in the cloud
+        expected_prefix="~{expected_output_prefix}"
+        # Remove a trailing slash if there is one
+        expected_prefix=${expected_prefix%/}
+
+        # Download all the expected data
+        mkdir expected
+        cd expected
+        # Make a FOFN for more efficient downloading
+        for file in ~{sep= ' ' actual_vcfs}; do
+            echo "$expected_prefix/$(basename $file)" >> expected_fofn.txt
+        done
+        # Download and unzip all the expected data
+        cat expected_fofn.txt | gsutil -m cp -I .
+        gzip -d *
+        cd ..
+
+        # Also unzip actual result data
+        gzip -d ~{sep= ' ' actual_vcfs}
+
+        # Headers first, these can yield useful diagnostics when there are mismatches.
+        for file in ~{sep=' ' actual_vcfs}; do
+          unzipped=${file%.gz}
+          expected="expected/$(basename $unzipped)"
+          cmp <(grep '^#' $unzipped) <(grep '^#' $expected)
+          if [[ $? -ne 0 ]]; then
+            # If there is a mismatch add it to a list of failures but keep on looking for mismatches.
+            failures+=( $unzipped )
+          fi
+        done
+
+        if [[ ${#failures[@]} -ne 0 ]]; then
+          echo "Error: headers for the following files do not match:"
+          for failure in ${failures[@]}; do
+            echo $(basename $failure)
+            expected="expected/$(basename $failure)"
+            diff $failure $expected
+          done
+          exit 1
+        fi
+
+        # If the headers all matched look for any mismatches in overall file content.
+        fail=0
+        for file in ~{sep=' ' actual_vcfs}; do
+          unzipped=${file%.gz}
+          expected="expected/$(basename $unzipped)"
+          cmp $unzipped $expected
+          if [[ $? -ne 0 ]]; then
+            echo "Error: file contents of expected and actual do not match: $(basename $unzipped)"
+            fail=1
+          fi
+        done
+
+        if [[ $fail -ne 0 ]]; then
+          exit 1
+        fi
+
+        echo "All vcfs compared and matched!"
+    >>>
```

Comment:
When we are ready to start doing fuzzy comparisons (with filtering), we could potentially use the ah_var_store docker image which has bcftools in it and do some easy selective comparisons that way. eg take two vcfs and compare only their variants as opposed to their bytes

Reply:
bcftools would also allow us to merge all of the shards into one fairly quickly which could also speed up the comparison